### PR TITLE
LibGfx: Clean up OpenType parsing by using C++ structs and BigEndian<T> a lot more

### PR DESCRIPTION
**Pull request by awesomekling**
Here's a first attempt to make our OpenType parser a nicer to work with.

### CODE REVIEW

File: Userland/Libraries/LibGfx/Font/OpenType/Tables.h [hunk 1]
```diff
@@ -219,6 +211,41 @@ class OS2 {
...
+        BigEndian<i16> y_subscript_y_offset;
+        BigEndian<i16> y_superscript_x_size;
+        BigEndian<i16> y_superscript_y_size;
+        BigEndian<i16> y_superscript_x_offset;
+        BigEndian<i16> y_superscript_y_offset;
+        BigEndian<i16> y_strikeout_size;
+        BigEndian<i16> y_strikeout_position;
+        BigEndian<i16> s_family_class;
+        u8 panose[10];
+        BigEndian<u32> ul_unicode_range1;
```
Comment (sunverwerth) [inline Userland/Libraries/LibGfx/Font/OpenType/Tables.h:233]: Do we need to pack this struct tightly? If I counted correctly `ul_unicode_range1` starts at byte offset 42.

Comment (awesomekling) [reply Userland/Libraries/LibGfx/Font/OpenType/Tables.h:233]: `BigEndian` is packed at the class level, so there should be no alignment padding inserted:
```cpp
template<typename T>
class [[gnu::packed]] BigEndian {
```

Comment (sunverwerth) [reply Userland/Libraries/LibGfx/Font/OpenType/Tables.h:233]: Oh didn't know that

**Review by GMTA — APPROVED** [review]
LGTM, seems to be working fine! Code is definitely nicer this way.

**Pull request merged**